src/key.js

```diff
@@ -491,11 +491,11 @@ function getExpirationTime(keyPacket, selfCertificate) {
 Key.prototype.getPrimaryUser = function() {
   var primUser = [];
   for (var i = 0; i < this.users.length; i++) {
-    if (!this.users[i].userId || !this.users[i].selfCertifications) {
+    if ((!this.users[i].userId && !this.users[i].userAttribute) || !this.users[i].selfCertifications) {
```

Comment:
Yeah, I will discrd this changes.
Anyway if it no primary user is found returns first valid self signed user